File: Remove loop.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
def remove_loop(head):
    slow, fast = head, head
    loop_found = False
    
    # Detect loop
    while fast and fast.next:
        slow = slow.next
        fast = fast.next.next
        if slow == fast:
            loop_found = True
            break
    
    if not loop_found:
        return
    
    # Move one pointer to head, keep other at meeting point
    slow = head
    while slow.next != fast.next:
        slow = slow.next
        fast = fast.next
    
    # Break loop
    fast.next = None

File: Remove loop.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->next = NULL;
    return node;
}
void removeLoop(struct Node* head) {
    struct Node* slow = head, *fast = head;
    bool loopFound = false;

    while (fast && fast->next) {
        slow = slow->next;
        fast = fast->next->next;
        if (slow == fast) { loopFound = true; break; }
    }
    if (!loopFound) return;

    slow = head;
    while (slow->next != fast->next) {
        slow = slow->next;
        fast = fast->next;
    }
    fast->next = NULL; // break loop
}
